orc8r/cloud/go/sqorc/builder.go

```diff
@@ -318,6 +318,26 @@ func ClearStatementCacheLogOnError(cache *squirrel.StmtCache, callsite string) {
 	}
 }
 
+// FmtConflictUpdateTarget generates the string used to refer to target value
+// used in an on-conflict upsert operation; the specific syntax depends on the
+// sql dialect.
+// e.g. for mariaDB, "f1=t2.f1"; for postgres, "f1=excluded.f1"
+func FmtConflictUpdateTarget(tableName string, colName string) string {
+	dialect, envFound := os.LookupEnv("SQL_DIALECT")
+	if !envFound {
+		dialect = PostgresDialect
+	}
+
+	var upsertColumnPrefix string
+	switch strings.ToLower(dialect) {
+	case PostgresDialect:
+		upsertColumnPrefix = "excluded"
+	case MariaDialect:
+		upsertColumnPrefix = tableName
+	}
```

Comment:
Always include a default in switch statements. In this case it's not strictly necessary, but it protects against future code changes

Since default here always means there's a problem, you can panic. Or if that feels icky, you can just return the postgres default